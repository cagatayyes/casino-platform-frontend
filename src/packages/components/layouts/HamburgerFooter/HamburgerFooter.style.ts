
import { StyledComponent } from 'casino/components/shared/StyledComponent';
import tw from 'tailwind-styled-components';

export const Container: StyledComponent = tw.div`
bg-[#0D1335]
bottom-[0px]
fixed
flex
w-full
height-[52px]
shadow-[0_2px_6px_0_#0D1335]
items-start
z-[1]
text-[11px]
text-[#687090]
max-w-screen-sm
self-center
`;
