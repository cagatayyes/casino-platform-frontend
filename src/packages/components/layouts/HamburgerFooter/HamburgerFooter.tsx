
import { Container } from './HamburgerFooter.style';
import { JSX } from 'react';
import Navigation from 'casino/components/Navigation/Navigation';
import { FooterNavigationItems } from 'casino/components/Navigation/Navigation.constants';
import { INavigationTriggerProps } from 'casino/components/Navigation/Navigation.type';

const HamburgerFooter = (
	{
		maxWidth,
		handleTabClick,
		activeTab,
	}: INavigationTriggerProps): JSX.Element => {

	return (
		<Container style={{ maxWidth }}>
			<Navigation
				options={FooterNavigationItems}
				classNames={'flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px]'}
				handleTabClick={handleTabClick}
				activeTab={activeTab}
			/>
		</Container>
	);
};

export default HamburgerFooter;
