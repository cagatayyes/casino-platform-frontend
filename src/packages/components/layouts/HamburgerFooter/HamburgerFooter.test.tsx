import { render } from '@testing-library/react';
import HamburgerFooter from './HamburgerFooter';
import React from 'react';

describe('Hamburger Footer', () => {
	it('should render Hamburger Footer', () => {
		const component = render(
			<HamburgerFooter
				maxWidth='640px'
				handleTabClick={jest.fn()}
				activeTab='OVERVIEW'
			/>,
		);
		expect(component).toMatchSnapshot();
	});
});
