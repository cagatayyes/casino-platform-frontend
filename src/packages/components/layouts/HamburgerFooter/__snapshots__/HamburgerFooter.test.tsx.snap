// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hamburger Footer should render Hamburger Footer 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="bg-[#0D1335] bottom-[0px] fixed flex w-full height-[52px] shadow-[0_2px_6px_0_#0D1335] items-start z-[1] text-[11px] text-[#687090] max-w-screen-sm self-center"
        style="max-width: 640px;"
      >
        <div
          class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] "
        >
          <span
            class="w-[16] h-[16] material-symbols-outlined"
            style="font-size: 16px;"
          >
            play_circle
          </span>
          <span
            class="text-[11px]"
          >
            RESUME
          </span>
        </div>
        <div
          class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] text-[#FFB800] bg-[#070C26] border-t-2 border-t-[#FFB800]"
        >
          <span
            class="w-[16] h-[16] material-symbols-outlined"
            style="font-size: 16px;"
          >
            tactic
          </span>
          <span
            class="text-[11px]"
          >
            OVERVIEW
          </span>
        </div>
        <div
          class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] "
        >
          <span
            class="w-[16] h-[16] material-symbols-outlined"
            style="font-size: 16px;"
          >
            more_horiz
          </span>
          <span
            class="text-[11px]"
          >
            MORE
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="bg-[#0D1335] bottom-[0px] fixed flex w-full height-[52px] shadow-[0_2px_6px_0_#0D1335] items-start z-[1] text-[11px] text-[#687090] max-w-screen-sm self-center"
      style="max-width: 640px;"
    >
      <div
        class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] "
      >
        <span
          class="w-[16] h-[16] material-symbols-outlined"
          style="font-size: 16px;"
        >
          play_circle
        </span>
        <span
          class="text-[11px]"
        >
          RESUME
        </span>
      </div>
      <div
        class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] text-[#FFB800] bg-[#070C26] border-t-2 border-t-[#FFB800]"
      >
        <span
          class="w-[16] h-[16] material-symbols-outlined"
          style="font-size: 16px;"
        >
          tactic
        </span>
        <span
          class="text-[11px]"
        >
          OVERVIEW
        </span>
      </div>
      <div
        class="flex flex-col items-center justify-center w-[25%] max-w-[25%] items-start cursor-pointer p-[3px] "
      >
        <span
          class="w-[16] h-[16] material-symbols-outlined"
          style="font-size: 16px;"
        >
          more_horiz
        </span>
        <span
          class="text-[11px]"
        >
          MORE
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
